### PR TITLE
[SP-5688] JAttach heapdump kubernetes step

### PR DESCRIPTION
**Pull request by moshebe**

### CODE REVIEW

File: steps/kubectl/jattach-heapdump/pod_files.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	remotePath string
+	pod        string
+	container  string
+	namespace  string
+}
+
+func (p podFiles) command(isDownload bool) string {
+	result := "cp "
+
+	podPath := p.pod + ":" + p.remotePath
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/pod_files.go:22]: I think fmt.sprintf would be cleaner

File: steps/kubectl/jattach-heapdump/storage.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+type storage struct {
+	Bucket string `env:"BUCKET,required"`
+}
+
+func (s storage) initGCloud() error {
+	// Note: our implementation injects the base64 representation of the credentials file from the integration.
+	// in order to seamlessly support it using go-cloud we will unpack the credentials to the expected format.
+	authCode := os.Getenv("AUTH_CODE")
+	if authCode == "" {
+		return nil
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/storage.go:23]:
```suggestion
		return fmt.Errorf("empty auth code provided")
```

File: steps/kubectl/jattach-heapdump/storage.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+type storage struct {
+	Bucket string `env:"BUCKET,required"`
+}
+
+func (s storage) initGCloud() error {
+	// Note: our implementation injects the base64 representation of the credentials file from the integration.
+	// in order to seamlessly support it using go-cloud we will unpack the credentials to the expected format.
+	authCode := os.Getenv("AUTH_CODE")
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/storage.go:21]: I think it should be in the configuration, and not in the manifest

File: steps/kubectl/jattach-heapdump/pod_files.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	if p.namespace != "" {
+		podPath = p.namespace + "/" + podPath
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/pod_files.go:24]: same with sprintf

File: steps/kubectl/jattach-heapdump/main.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	if err != nil {
+		return fmt.Errorf("execute command due to %s: %w", string(out), err)
+	}
+	m.kubectl.Debugln(string(out))
+	return nil
+}
+
+func (m *memoryDump) dump() error {
+	return m.exec(
+		fmt.Sprintf("chmod +x %s && %s", m.JAttach.DumpScriptRemotePath, m.JAttach.DumpScriptRemotePath),
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/main.go:154]: use const for the command

File: steps/kubectl/jattach-heapdump/jattach.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+const (
+	dumpScriptLocalPath = "/tmp/jattach-dump.sh"
+	pulledDumpLocalPath = "/tmp/jattach-heapdump.hprof"
+	localJAttachPath    = "/jattach"
+)
+
+func (j jattach) WriteDumpScript(writer io.Writer) error {
+	commands := []string{
+		fmt.Sprintf("chmod +x %s", j.RemotePath),
+		fmt.Sprintf("%s $(pidof -s java) dumpheap %s", j.RemotePath, j.OutputDumpPath),
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/jattach.go:24]: use const

File: steps/kubectl/jattach-heapdump/jattach.go [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+const (
+	dumpScriptLocalPath = "/tmp/jattach-dump.sh"
+	pulledDumpLocalPath = "/tmp/jattach-heapdump.hprof"
+	localJAttachPath    = "/jattach"
+)
+
+func (j jattach) WriteDumpScript(writer io.Writer) error {
+	commands := []string{
+		fmt.Sprintf("chmod +x %s", j.RemotePath),
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/jattach.go:23]: use const

File: steps/kubectl/jattach-heapdump/main.go [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+	return nil
+}
+
+func (m *memoryDump) cleanup() error {
+	files := []string{
+		m.JAttach.DumpScriptRemotePath,
+		m.JAttach.OutputDumpPath,
+		m.JAttach.RemotePath,
+	}
+	return m.exec(fmt.Sprintf("rm -f %s", strings.Join(files, " ")))
```
Comment (talbe) [inline steps/kubectl/jattach-heapdump/main.go:184]: use const

**Review by talbe — APPROVED** [review]

**Comment by talbe** [comment]
LGTM

**Review by hagaishapira — APPROVED** [review]
LGTM

**Pull request merged**